### ISSUE TITLE
[FR] Better support for non-ASCII text on Windows (RStudio Desktop)

### ISSUE THREAD

**Issue by shrektan** [rstudio/rstudio#5684]
First of all, what I describe only occurs on Windows.

## Issue Description

RStudio has alread done a good job of supporting non-ASCII strings. However, there're still some issues. Here's a list of existing encoding issues that I encountered (I will update when I can think of more):

1. Code completion breaks whenever the package document on that function contains multi-bytes text. Apparently, it's because the R code completion will display the docs on that function while RStudio can't display that non-ASCII strings.

1. (may have been solved in v1.3.X. will confirm) The "Terminal" panel will display any non-ASCII messages as raw UTF-8 values,  like "<U+8xxx>", maybe an issue about fonts?

1. The default encoding for the scripts on Windows is still "[ASK]" . It will save many R users' time if this is "UTF-8" by default.

1. ~~The "file path auto-completion"  can't work properly when the path contains non-ASCII strings.~~(2019.11.04: Sorry for my bad memory. I checked and this works fine)

1. It will be great if a UTF-8 BOM(Byte order mark) can be attached in the CSV file (if we enabled this function), whenever it gets saved. It's because edit some content in Excel is far better than edit the raw text. Without the BOM, opening a CSV file with UTF-8 content in Excel leads to garbage (Yes I know Excel should have detected it's a UTF-8 CSV but it just doesn't). However, I'm not sure if there's any side-effect by adding a UTF-8 BOM. If not, then it seems a good idea to me.

1. Some functions in `rstudioapi` call RStudio's C++ functions, which don't support nonASCII strings well on Windows. For example, `rstudioapi::showQuestion()` https://github.com/rstudio/rstudioapi/issues/159

1. Whenever the data.frame or other objects contain multi-bytes strings, it can't be displayed with good format. I mean, it can't be aligned properly. It may be an issue about fonts as well.

    ```r
    # this is a simple illustration, in reality, when there're many CN strings
    # the column X's content may be displayed in column Y. Making recognize
    # the content difficult. 
    tibble::tibble(
      a = c("中文", "aaaaaa"),
      b = c("中文", "aaaaaa")
    )
    ```

## System Info

- Windows computer with any multi-bytes locale (e.g., GB2312 for Chinese)
- Any RStudio Desktop version

**Comment by gtritchie** [comment]
The terminal _should_ be better behaved in 1.3 due to an upgrade of the underlying component (xterm.js). That said, I haven't confirmed this.

**Comment by shrektan** [comment]
I confirmed this one month ago when it was still 1.2.X. I will test this next week and remove this line if it has been solved already. Thanks.

**Comment by shrektan** [comment]
@gtritchie I still can't print Chinese characters with the lastet RStudio daily builds on Windows 7.

# On jobs I get gabarge letters like ???

![image](https://user-images.githubusercontent.com/8368933/68926111-ad8bf500-07bf-11ea-879f-932c245b7271.png)

# On terminal I get empty for Chinese characters

![image](https://user-images.githubusercontent.com/8368933/68926190-e2984780-07bf-11ea-97c6-6979e1d4bbf5.png)

# RStudio Version

![image](https://user-images.githubusercontent.com/8368933/68926215-f2179080-07bf-11ea-98b8-eff8059b6c30.png)

**Comment by jmcphers** [comment]
Appreciate you taking the time to summarize these, but as they're all relatively unrelated issues we will address them one at a time rather than as a unit. Several of these are already fixed in 1.3 and others are tracked as separate issues (e.g. RStudio Desktop on Windows: wrong encoding in popup help during code completion). Feel free to open other issues individually with reproducible examples if you've got them!
